src/main/java/ViewModel/uploadSongViewModel.java

```diff
@@ -1,4 +1,33 @@
 package ViewModel;
+import Controller.uploadSongController;
+import InputData.uploadSongInputData;
+import Presenter.uploadSongPresenter;
 
 public class uploadSongViewModel {
+
+    private final uploadSongController uploadSongController;
+    private boolean isUploaded;
+
+    uploadSongViewModel(){
+        this.uploadSongController = new uploadSongController(new uploadSongPresenter(this));
+    }
+
+    /**
+     *
+     * @param filepath Filepath to the desired song.
+     * @param user The username of the uploading user.
+     */
+    public void upload(String filepath, String user){
+        uploadSongInputData inputdata = new uploadSongInputData(filepath, user);
+        uploadSongController.upload(inputdata);
```

Comment:
might have to return isUploaded to the View -- not sure what your implementation plans were for the View

Reply:
Ah true. Haven't looked at view yet, probably why i didn't write it.